### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by dsounded**
https://app.asana.com/0/156934457071997/158849269265978

In this request:

1) added custom aciton `AuthorizationAction` which include custom header based auth, which also add user object to request.

2)Provided example GET `/countries`, let's decide which actions should we cover with Authentication.

3) Provided example of documenting for `swagger`.

4) Provided tests with `auth` header

### CODE REVIEW

**Comment by dsounded** [comment]
I've branched from `swagger request`, so I'll squash commits after api being merged into master

File: app/actions/AutorizationAction.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    getUser(header) flatMap {
+      case (user, error) if error isEmpty => block(AuthenticatedRequest(user, request))
+      case (_, error)                     => Future.successful(Status(responder.Forbidden)(Json.obj(responder.ErrorRootName -> Json.toJson(error))))
+    }
+  }
+
+  def getUser(header: String) = {
+    Users.findByAuthToken(header) map {
+      case Some(user) => {
```
Comment (hydro0) [inline app/actions/AutorizationAction.scala]: case Some(user) if Receptionist.canEnter(user) => (user, Set.empty)
case Some(user) => (user, responser.ExpiredTokenError)

IMHO looks better.

File: app/utils/Time.scala [hunk 2]
```diff
@@ -4,4 +4,10 @@ import java.sql.Timestamp
 
 object TimeUtil {
   def now = new Timestamp(System.currentTimeMillis)
+
+  def addDays(startDate: Timestamp, days: Int): Timestamp = {
+    new Timestamp(startDate.getTime + days * 3600 * 24 * 1000)
+  }
+
+  def moreThanNow(date: Timestamp): Boolean = date.getTime >= now.getTime
```
Comment (hydro0) [inline app/utils/Time.scala]: date.after(now)

**Comment by hydro0** [comment]
Ship it!

**Pull request merged**